pkg/gui/files_panel.go

```diff
@@ -211,6 +211,23 @@ func (gui *Gui) handleCommitPress(g *gocui.Gui, filesView *gocui.View) error {
 	return nil
 }
 
+func (gui *Gui) handleAmendCommitPress(g *gocui.Gui, filesView *gocui.View) error {
+	if len(gui.stagedFiles()) == 0 && !gui.State.HasMergeConflicts {
+		return gui.createErrorPanel(g, gui.Tr.SLocalize("NoStagedFilesToCommit"))
+	}
+	title := strings.Title(gui.Tr.SLocalize("AmendLastCommit"))
+	question := gui.Tr.SLocalize("SureToAmend")
+	return gui.createConfirmationPanel(g, filesView, title, question, func(g *gocui.Gui, v *gocui.View) error {
+		lastCommitMsg := gui.State.Commits[0].Name
```

Comment:
it is possible that we have just done a `git init` so there are no commits yet. I would chuck a check for that here:
```
if len(gui.State.Commits) == 0 {
    return errors.New(gui.Tr.SLocalize("NoCommitToAmend"))
}
```

Reply:
Should i do it before the confirmation panel, or inside of it?

Reply:
Done.